### PR TITLE
Refactor sign in after co tests

### PR DESCRIPTION
**Pull request by lindseydew**
## What does this change?
This moves the testing of the variations of the types of user and product a user may have when they have the `GU_CO_COMPLETE` cookie to test at the component level, rather than using e2e tests. 
## Why?
E2E tests take longer to run and harder to debug, and so these are tests are more appropriate at the component leve. 

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |

[before]: https://example.com/before.png
[after]: https://example.com/after.png

### CODE REVIEW

**Comment by mxdvl** [comment]
Mentioning @arelra who has been moving these Cypress tests to Playwright

**Review by arelra — APPROVED** [review]
Great to see tests move from e2e to component level tests!

> E2E tests take longer to run and harder to debug

Hopefully the current migration to Playwright should mean e2e tests are easier to debug using the [VS Code plugin](https://marketplace.visualstudio.com/items?itemName=ms-playwright.playwright).

If it helps anyone else - e2e tests are useful when we want to test in a real browser, or for cross component behaviour or more involved user interaction flows.

But where tests can be isolated to just the component we should prefer that as done here - thank you @lindseydew !

File: dotcom-rendering/src/lib/signInAfterCheckOutText.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// HEADER TEXT
+export const SUBSCRIPTION_HEADER = 'Thank you for subscribing';
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/signInAfterCheckOutText.ts:2]: If you use JSDoc you’ll get nice IDE integration

```suggestion
/** Header text */
export const SUBSCRIPTION_HEADER = 'Thank you for subscribing';
```

Comment (lindseydew) [reply dotcom-rendering/src/lib/signInAfterCheckOutText.ts:2]: Oh nice, what's JSDoc?

Comment (lindseydew) [reply dotcom-rendering/src/lib/signInAfterCheckOutText.ts:2]: Oh, literally what you have typed out. Got it 😅

**Pull request merged**